akka-persistence/src/main/scala/akka/persistence/Persistent.scala

```diff
@@ -41,7 +41,7 @@ final case class AtomicWrite(payload: immutable.Seq[PersistentRepr]) extends Per
 
   // only check that all persistenceIds are equal when there's more than one in the Seq
   if (payload match {
-    case l: List[PersistentRepr]   ⇒ l.tail.nonEmpty
+    case l: List[PersistentRepr]   ⇒ l.nonEmpty && l.tail.nonEmpty // avoids calling .size
```

Comment:
Looks like `AtomicWrite` is not really designed for empty `payload`, there are problems in `lowestSequenceNr` and `highestSequenceNr` also. I don't mind if we we use it internally for looping to the journal, but should avoid calling `asynWriteMessages` with empty `AtomicWrite` ?